docs/sources/source/nlp/joint_intent_slot_filling.rst

```diff
@@ -148,85 +143,73 @@ Next, we define all Neural Modules participating in our joint intent slot fillin
             steps_per_epoch = math.ceil(data_size / (batch_size * num_gpus))
             logging.info(f"Steps_per_epoch = {steps_per_epoch}")
 
-            hidden_states = pretrained_bert_model(input_ids=ids,
-                                                  token_type_ids=type_ids,
-                                                  attention_mask=input_mask)
+            hidden_states = pretrained_bert_model(input_ids=ids, token_type_ids=type_ids, attention_mask=input_mask)
 
             intent_logits, slot_logits = classifier(hidden_states=hidden_states)
 
-            loss = loss_fn(intent_logits=intent_logits,
-                           slot_logits=slot_logits,
-                           loss_mask=loss_mask,
-                           intents=intents,
-                           slots=slots)
+            intent_loss = intent_loss_fn(logits=intent_logits, labels=intents)
+            slot_loss = slot_loss_fn(logits=slot_logits, labels=slots, loss_mask=loss_mask)
+            total_loss = total_loss_fn(loss_1=intent_loss, loss_2=slot_loss)
 
             if mode == 'train':
-                tensors_to_evaluate = [loss, intent_logits, slot_logits]
+                tensors_to_evaluate = [total_loss, intent_logits, slot_logits]
             else:
-                tensors_to_evaluate = [intent_logits, slot_logits, intents,
-                                       slots, subtokens_mask]
+                tensors_to_evaluate = [intent_logits, slot_logits, intents, slots, subtokens_mask]
 
-            return tensors_to_evaluate, loss, steps_per_epoch, data_layer
+            return tensors_to_evaluate, total_loss, steps_per_epoch, data_layer
 
 
         train_tensors, train_loss, steps_per_epoch, _ = create_pipeline(
-            args.num_train_samples,
-            batch_size=args.batch_size,
-            num_gpus=args.num_gpus,
-            local_rank=args.local_rank,
-            mode=args.train_file_prefix)
-        eval_tensors, _,  _, data_layer = create_pipeline(
-            args.num_eval_samples,
-            batch_size=args.batch_size,
-            num_gpus=args.num_gpus,
-            local_rank=args.local_rank,
-            mode=args.eval_file_prefix)
+            args.num_train_samples, batch_size=args.batch_size, num_gpus=args.num_gpus, mode=args.train_file_prefix,
+        )
+        eval_tensors, _, _, data_layer = create_pipeline(
+            args.num_eval_samples, batch_size=args.batch_size, num_gpus=args.num_gpus, mode=args.eval_file_prefix,
+        )
 
     * Create relevant callbacks for saving checkpoints, printing training progresses and evaluating results.
 
     .. code-block:: python
 
-        train_callback = nemo.core.SimpleLossLoggerCallback(
+        train_callback = SimpleLossLoggerCallback(
```

Comment:
should we in the tutorial describe how to import these Classes? in that case nemo.core.SimpleLossLoggerCallback makes more sense

Reply:
In tutorial we actually just explain what happens in the example. So they should check the example to understand whats happening otherwise no way. How to do that? Add the imports to the tutorial?

Reply:
i have added the imports for other tutorials

Reply:
Done.